src/transformers/generation_utils.py

```diff
@@ -870,16 +946,18 @@ def generate(
             >>> outputs = model.generate(input_ids=input_ids, max_length=20, do_sample=True, bad_words_ids=bad_words_ids)
             >>> print("Generated:", tokenizer.decode(outputs[0], skip_special_tokens=True))
         """
-
+        # 1. Set generation parameters if not already defined
+        bos_token_id = bos_token_id if bos_token_id is not None else self.config.bos_token_id
```

Comment:
No the goal of this PR, but maybe we can think of doing something like.

```python
config_arguments = ["bos_token_id", .....]
for argument in config_arguments:
    locals()[argument] = locals().get(arguments, getattr(self.config, argument))
```

This is not exactly cool to have such code, but the clutter of all the other arguments (and apparently some missing ones) might make it worth it.

We might even fetch `config_arguments` from the `Config` class itself (if relevant) so we don't forget to add them here when adding config level generate arguments.